File: repo/frontend/src/api/content.ts
// API client for governed content.

import { apiClient } from './client';
import type { PageResult } from './readers';

export type ContentStatus = 'draft' | 'pending_review' | 'approved' | 'rejected' | 'published' | 'archived';
export type ContentType = 'announcement' | 'document' | 'digital_resource' | 'policy';

export interface GovernedContent {
  id: string;
  branch_id: string;
  title: string;
  content_type: ContentType;
  body?: string;
  file_name?: string;
  status: ContentStatus;
  submitted_by: string;
  submitted_at: string;
  published_at?: string;
  archived_at?: string;
  rejection_reason?: string;
  created_at: string;
  updated_at: string;
}

function listContent(params?: {
  status?: string;
  content_type?: string;
  page?: number;
  per_page?: number;
}): Promise<PageResult<GovernedContent>> {
  const qs = new URLSearchParams();
  if (params?.status) qs.set('status', params.status);
  if (params?.content_type) qs.set('content_type', params.content_type);
  if (params?.page) qs.set('page', String(params.page));
  if (params?.per_page) qs.set('per_page', String(params.per_page));
  const q = qs.toString();
  return apiClient.get<PageResult<GovernedContent>>(`/content${q ? '?' + q : ''}`);
}

function getContent(id: string): Promise<GovernedContent> {
  return apiClient.get<GovernedContent>(`/content/${id}`);
}

function createContent(data: {
  title: string;
  content_type: ContentType;
  body?: string;
  file_name?: string;
}): Promise<GovernedContent> {
  return apiClient.post<GovernedContent>('/content', data);
}

function updateContent(id: string, data: Partial<{
  title: string;
  body: string;
  file_name: string;
}>): Promise<GovernedContent> {
  return apiClient.patch<GovernedContent>(`/content/${id}`, data);
}

function submitContent(id: string): Promise<GovernedContent> {
  return apiClient.post<GovernedContent>(`/content/${id}/submit`, {});
}

function retractContent(id: string): Promise<GovernedContent> {
  return apiClient.post<GovernedContent>(`/content/${id}/retract`, {});
}

function publishContent(id: string): Promise<GovernedContent> {
  return apiClient.post<GovernedContent>(`/content/${id}/publish`, {});
}

function archiveContent(id: string): Promise<GovernedContent> {
  return apiClient.post<GovernedContent>(`/content/${id}/archive`, {});
}

export const contentApi = {
  list: listContent,
  get: getContent,
  create: createContent,
  update: updateContent,
  submit: submitContent,
  retract: retractContent,
  publish: publishContent,
  archive: archiveContent,
};

File: repo/backend/internal/model/content.go
package model

import "time"

// GovernedContent is a library-published item that goes through moderation
// before becoming visible to readers.
// Lifecycle: draft → pending_review → (approved|rejected) → published → archived
type GovernedContent struct {
	ID              string     `json:"id"`
	BranchID        string     `json:"branch_id"`
	Title           string     `json:"title"`
	ContentType     string     `json:"content_type"` // announcement, document, digital_resource, policy
	Body            *string    `json:"body,omitempty"`
	FilePath        *string    `json:"-"`          // internal filesystem path; not exposed in API
	FileName        *string    `json:"file_name,omitempty"`
	Status          string     `json:"status"` // draft, pending_review, approved, rejected, published, archived
	SubmittedBy     string     `json:"submitted_by"`
	SubmittedAt     time.Time  `json:"submitted_at"`
	PublishedAt     *time.Time `json:"published_at,omitempty"`
	ArchivedAt      *time.Time `json:"archived_at,omitempty"`
	RejectionReason *string    `json:"rejection_reason,omitempty"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
}

// ModerationItem is a queue entry for content awaiting moderation review.
type ModerationItem struct {
	ID             string     `json:"id"`
	ContentID      string     `json:"content_id"`
	AssignedTo     *string    `json:"assigned_to,omitempty"`
	Status         string     `json:"status"`   // pending, in_review, decided
	Decision       *string    `json:"decision,omitempty"`        // approved, rejected
	DecisionReason *string    `json:"decision_reason,omitempty"`
	DecidedBy      *string    `json:"decided_by,omitempty"`
	DecidedAt      *time.Time `json:"decided_at,omitempty"`
	CreatedAt      time.Time  `json:"created_at"`
	UpdatedAt      time.Time  `json:"updated_at"`
}
